packages/graphql/src/utils/execute.ts

```diff
@@ -150,7 +183,9 @@ async function execute(input: {
 
         throw error;
     } finally {
-        await session.close();
+        if (executor.driver && executor.session) {
```

Comment:
Why checking for `executor.driver` here if this code is only closing the session?

Reply:
I realize this code is not very intuitive. We have two scenarios here:

1. If a session was provided via executionContext, we do not want to automatically close that session.
2. If no session was provided via executionContext, it means that the session was created by the execute function, and thus should be closed automatically.

If a driver was passed either via context.driver, or context.executionContext it means that the session was created by the execute function and should be automatically closed. There are probably better / more intuitive ways of achieving this.

Reply:
I seeeee, may I suggest a boolean variable to describe this behaviour?

```suggestion
        const sessionCreatedByExecute = executor.driver && executor.session;
        if (sessionCreatedByExecute) {
```

Reply:
Sounds good, but we might want to cast it too? 

`Boolean(executor.driver && executor.session)`